src/components/nabla/Swap/index.tsx

```diff
@@ -19,28 +18,27 @@ const Swap = (props: UseSwapComponentProps): JSX.Element | null => {
     swapMutation,
     onSubmit,
     form,
-    from,
     updateStorage,
     progressClose,
-    tokensQuery,
+    swapPools,
+    nablaInstanceIsLoading,
+    fromTokenBalance,
+    fromAmountString,
+    fromAmount,
+    toAmountQuote,
+    fromToken,
+    toToken,
+    slippage,
   } = useSwapComponent(props);
+
   const {
     setValue,
     register,
-    getValues,
     formState: { errors },
   } = form;
-  const { tokens, tokensMap } = tokensQuery.data || {};
 
-  const progressUi = useMemo(() => {
-    if (swapMutation?.isIdle) return null;
-    const { from: fromV, to: toV, fromAmount = 0, toAmount = 0 } = getValues();
-    const fromAsset = tokensMap?.[fromV];
-    const toAsset = tokensMap?.[toV];
-    return (
-      <p className="text-center text-[--text]">{`Swapping ${fromAmount} ${fromAsset?.symbol} for ${toAmount} ${toAsset?.symbol}`}</p>
-    );
-  }, [getValues, swapMutation?.isIdle, tokensMap]);
+  const submitEnabled = !toAmountQuote.isLoading && toAmountQuote.enabled && Object.keys(errors).length === 0;
```

Comment:
I think instead of using `Object.keys(errors).length === 0`, we can use isValid from `formState: { errors, isValid }`

Reply:
See my other comment.